homeassistant/components/volvooncall/__init__.py

```diff
@@ -129,13 +195,13 @@ def discover_vehicle(self, vehicle):
             if instrument.component in PLATFORMS
         ):
             self.instruments.add(instrument)
+            dispatcher_send(self.hass, VOLVO_DISCOVERY_NEW, [instrument])
 
-        for platform in PLATFORMS:
-            self.hass.async_create_task(
-                self.hass.config_entries.async_forward_entry_setup(
-                    self.config_entry, platform
-                )
+        self.hass.async_create_task(
+            self.hass.config_entries.async_forward_entry_setups(
+                self.config_entry, PLATFORMS
             )
+        )
```

Comment:
We should move this to the config entry setup and await it after the first coordinator update. The platforms will create entities for the registered vehicles during platform set up so it's ok if we miss the first discovery dispatch.
```py
await self.hass.config_entries.async_forward_entry_setups(
    config_entry, PLATFORMS
)
```